sdk/go/common/workspace/plugins.go

```diff
@@ -1806,19 +1806,34 @@ func getPluginInfoAndPath(
 						(stat.Mode()&0o100 != 0 || runtime.GOOS == windowsGOOS) {
 						logging.V(6).Infof("GetPluginPath(%s, %s, %v): found next to current executable %s",
 							kind, name, version, candidate)
-
-						return &PluginInfo{
-							Kind: kind,
-							Name: name,
-							Path: filepath.Dir(candidate),
-						}, candidate, nil
+						bundledPath = candidate
+						break
 					}
 				}
 			}
 		}
 	}
 
-	// Otherwise, check the plugin cache.
+	// We prefer the ambient path, but we need to check if this is the same as the bundled
+	// path to decide if we're warning or not.
+	if ambientPath != "" {
+		if ambientPath != bundledPath {
```

Comment:
don't you need to check for `bundledPath != ""` here?

Reply:
Nope, if bundledPath is empty then this was a non-bundled plugin (like a provider) and we want to warn. `"path" != ""` is true and we log the warning.